### ISSUE TITLE
Some problems about dimension with Pad Layers

### ISSUE THREAD

**Issue by PhyllisJi** [pytorch/pytorch#105627]
### 🐛 Describe the bug

When using some layers that internally call F.pad, these layers appear to have loose distinctions regarding dimensions, including distinctions for parameter "padding" and distinctions for input tensor dimensions.

These layers, including ConstantPad, ZeroPad, ReflectionPad and ReplicationPad(and their 1d 2d 3d), exhibit the aforementioned loose distinctions regarding dimensions when called in the program. This is because they do not check the input tensor and input parameters but directly call F.pad for processing.

### 1. "padding" dimensions
Here is an example to show a problem about "padding"

input:
```
p = nn.ZeroPad2d(padding=(1,2))
x = torch.randn(3,3,32,32)
y = p(x)
print(y.shape)
```
output:
```
torch.Size([3, 3, 32, 35])
```
In this case, the interface requires the parameter to be **int** or **tuple[int, int, int, int]**, and the functionality is to perform extension on the last two dimensions in the upward, downward, left, and right directions(which are shown in doc and examples), but an actual tuple with a length of 2 is passed. Due to the lack of checks, this parameter can run successfully, and its functionality extends the last dimension in both left and right directions, which does not comply with the original specification of the interface. 

Similarly, in other Paddings and each of their 1d/2d/3d interface, this problem also exists.

### 2. Input dimensions
An example is as below:
input:
```
p = nn.ConstantPad3d(padding=1, value=0)
x = torch.randn(3,3,32,)
y = p(x)
print(y.shape)
```
output:
```
torch.Size([5, 5, 34])
```
In this example, p is defined as a 3D extension layer, but when a three-dimensional tensor is input (which should have used a 1D extension layer), the 3D layer processes it and also extends the batch_size and channels of the tensor. This issue still occurs in all of these Pad interfaces.

### Versions

PS D:\PythonProjects\venv\Lib\site-packages\torch\utils> python collect_env.py
Collecting environment information...
PyTorch version: 2.0.1+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 家庭中文版
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.22621-SP0
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 4070 Laptop GPU
Nvidia driver version: 532.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture=9
CurrentClockSpeed=2200
DeviceID=CPU0
Family=207
L2CacheSize=16384
L2CacheSpeed=
Manufacturer=GenuineIntel
MaxClockSpeed=2200
Name=13th Gen Intel(R) Core(TM) i9-13900HX
ProcessorType=3
Revision=

Versions of relevant libraries:
[pip3] numpy==1.24.3
[pip3] torch==2.0.1+cu117
[pip3] torchaudio==2.0.2+cu117
[pip3] torchvision==0.15.2+cu117
[conda] Could not collect

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

**Comment by mikaylagawarecki** [comment]
Hi @PhyllisJi, you are right that all these classes subclass `_ConstantPadNd` or `_ReflectionPadNd` or `_ReplicationPadNd` which call ` F.pad(input, self.padding, {mode})` and do not seem to 

re:

(1) The classes have proper type hints for the `padding` input but do not have explicit checks for shape. I think an actionable fix here would be to update [`nn.utils._ntuple`](https://github.com/pytorch/pytorch/blob/main/torch/nn/modules/utils.py#L8) to raise an error if the Iterable passed is len != n

(2) It does seem like the code is not checking the input shape. Some other follow the same convention of have 1d, 2d and 3d variants that inherit from `_{*}Nd` classes that call the functional variant but have proper input shape checking. An example is [`_check_input_dim`](https://github.com/pytorch/pytorch/blob/02cd971e958f4e2f9073e5021501e88cd0460731/torch/nn/modules/batchnorm.py#L299) in BatchNorm which is [called in the forward of the base class](https://github.com/pytorch/pytorch/blob/02cd971e958f4e2f9073e5021501e88cd0460731/torch/nn/modules/batchnorm.py#L138) It would be actionable to add these checks in a consistent manner to the padding variants as well.

**Comment by PhyllisJi** [comment]
> Hi @PhyllisJi, you are right that all these classes subclass `_ConstantPadNd` or `_ReflectionPadNd` or `_ReplicationPadNd` which call ` F.pad(input, self.padding, {mode})` and do not seem to have proper error checking for these cases.
> 
> re:
> 
> (1) The classes have proper type hints for the `padding` argument but do not have explicit checks for shape. I believe an actionable fix here would be to update [`nn.utils._ntuple`](https://github.com/pytorch/pytorch/blob/main/torch/nn/modules/utils.py#L8) to raise an error if the Iterable passed has`len != n`
> 
> (2) It does seem like the code is not checking the input shape. Some other modules follow the same convention of have 1d, 2d and 3d variants that inherit from `_{*}Nd` classes that call the functional variant but have proper input shape checking. An example is [`_check_input_dim`](https://github.com/pytorch/pytorch/blob/02cd971e958f4e2f9073e5021501e88cd0460731/torch/nn/modules/batchnorm.py#L299) in BatchNorm which is [called in the forward of the base class](https://github.com/pytorch/pytorch/blob/02cd971e958f4e2f9073e5021501e88cd0460731/torch/nn/modules/batchnorm.py#L138) It would be actionable to add these checks in a consistent manner to the padding variants as well.

Thank you for your reply! Sounds like a comprehensive solution
